### PR TITLE
Added simple support for subnets (EC2)

### LINKED ISSUE #13: VPC support for EC2
Internal ID: 20

EC2 node should have configurable VPC. Sth like:
* use existing subnet ID
* create a new subnet

### PR DESCRIPTION
**Pull request by LittleJohnII**
Fixes #13 , at least partially.

It is now possible to use subnets with the EC2 provider by specifying subnet ID and security group IDs. For now, if you want to use subnets, you need to specify its ID. This could be improved as part of this PR or later (your pick, I don't need that functionality for now).

Specifying security group IDs is unfortunately necessary due to https://issues.apache.org/jira/browse/JCLOUDS-1120.

The test passes for me, let me know if you need proper `subnetId` and `securityGroupIDs` for testing.

### CODE REVIEW

File: core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Starts a node, then queries the AWS EC2 API to find out if the node really belongs to the
+     * subnet that was specified in the .properties file.
+     */
+    @Test
+    public void subnetTest() {
+        CloudProperties.getInstance().reset().load(EC2SubnetTest.class);
+        try (EC2CloudProvider ec2CloudProvider = (EC2CloudProvider) CloudProvider.create("provider0")) {
+            EC2Node ec2Node = (EC2Node) ec2CloudProvider.createNode("node0");
```
Comment (Ladicek) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java]: This node is not destroyed.

Comment (LittleJohnII) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java]: It is, when the ec2CloudProvider is closed. From logs:
```
...
09:08:57.986 [main] INFO  sunstone.core - Destroying Amazon EC2 cloud provider 'provider0'
09:09:29.247 [main] INFO  sunstone.core - Destroyed Amazon EC2 node 'node0'
09:09:29.248 [main] INFO  sunstone.core - Destroyed Amazon EC2 cloud provider 'provider0'
```

Comment (Ladicek) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java]: Ah, true, I forgot that closing the cloud provider closes all non-closed nodes.

I wonder why we always write `try (Node node = cloudProvider.createNode(...)) { ... }` when it really isn't necessary.

Comment (kwart) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java]: It's cleaner solution to close the Node explicitly. The developer declares - I don't need this VM anymore.

File: core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.junit.Test;
+import org.wildfly.extras.sunstone.api.CloudProperties;
+import org.wildfly.extras.sunstone.api.CloudProvider;
+import org.wildfly.extras.sunstone.api.impl.Config;
+import org.wildfly.extras.sunstone.api.impl.ec2.EC2CloudProvider;
+import org.wildfly.extras.sunstone.api.impl.ec2.EC2Node;
+
+import java.util.Set;
+
+public class EC2SubnetTest {
```
Comment (kwart) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java]: Add some documentation here, please. What it does and what's necessary to configure to run the tests?

**Comment by kwart** [comment]
I'm missing documentation coverage in this PR.

Could you cover the `CHANGELOG.md`, `ec2-README.md` and probably the `CONTRIBUTING.md` too?

Otherwise LGTM.

File: core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java (hunk 1 above)
Comment (LittleJohnII) [inline core/src/test/java/org/wildfly/extras/sunstone/tests/ec2/EC2SubnetTest.java]: Makes sense, I'll fix it.

**Comment by LittleJohnII** [comment]
Pffft. Documentation is overrated... (it slipped my mind, sorry)

Fixed it.

**Comment by kwart** [comment]
Thx for fixes. Let's squash and merge.

**Comment by LittleJohnII** [comment]
Squashed.

**Comment by kwart** [comment]
Merged.

**Pull request merged**